src/import_whatsapp.js

```diff
@@ -235,57 +277,89 @@ const processText = (text) => {
 		features: [],
 	};
-	let feature = null;
+	let currentFeature = null;
 	let currentSender = null;
 
+	const createFeature = (message, groupName) => {
+		return {
+			type: "Feature",
+			properties: {
+				contributionid: crypto.randomUUID(),
+				mainattribute: groupName,
+				observations: "",
+				observer: message.sender,
+				datetime: message.datetime,
+				markerColour: senders[message.sender],
+				imgFilenames: [],
+			},
+			geometry: message.location
+				? {
+					type: "Point",
+					coordinates: [message.location.long, message.location.lat],
+				}
+				: null,
+		};
+	};
+
+	const isValidContent = (content) => {
+		content = content.replace(
+			"Messages and calls are end-to-end encrypted. No one outside of this chat, not even WhatsApp, can read or listen to them.",
+			""
+		);
+		content = content.replace("<This message was edited>", "");
+
+		return content
+			.split("\n")
+			.map((line) => line.trim())
+			.filter(Boolean)
+			.join("\n");
+	};
+
 	messages.forEach((message) => {
-		// Sometimes sender changes but there's no location so add any features in progress
-		if (message.sender != currentSender || message.location) {
-			// Push any existing features to mapdata
-			if (feature) {
-				// Reject feature if it doesn't have geometry
-				if (feature.geometry) {
-					mapdata.features.push(feature);
-				} else {
-					feature = null;
+		// if the content is valid and there is location or different sender, get the current feature or create a new one and push it to mapdata
+		if (isValidContent(message.content)) {
+			if (message.location || message.sender !== currentSender) {
+				if (currentFeature && currentFeature.geometry) {
+					mapdata.features.push(currentFeature);
 				}
+				currentFeature = createFeature(message, groupName);
+				currentSender = message.sender;
 			}
-			// Create new feature
-			var contributionid = crypto.randomUUID();
-			feature = {
-				type: "Feature",
-				properties: {
-					contributionid: contributionid,
-					mainattribute: groupName,
-					observations: "",
-					observer: message.sender,
-					datetime: message.datetime,
-					markerColour: senders[message.sender],
-				},
-			};
-			if (message.location) {
-				feature.geometry = {
-					type: "Point",
-					coordinates: [message.location.long, message.location.lat], // GeoJSON uses [long, lat] order
-				};
-			}
-		} else if (feature) {
-			// Append message content to observations unless it's media omitted
-			if (!message.content.includes("<Media omitted>")) {
-				feature.properties.observations += message.content + "\n";
+
+			if (currentFeature) {
+				if (message.imgFilenames) {
+					currentFeature.properties.imgFilenames.push(...message.imgFilenames);
+				}
+				currentFeature.properties.observations += message.content + "\n";
 			}
 		}
-		currentSender = message.sender;
 	});
 	// Push the last message to mapdata
-	if (feature) {
-		mapdata.features.push(feature);
+	if (currentFeature) {
```

Comment:
I think unless we check the last feature has geometry, our no locations in chat thing won't work.
```suggestion
	if (currentFeature && currentFeature.geometry) {
```